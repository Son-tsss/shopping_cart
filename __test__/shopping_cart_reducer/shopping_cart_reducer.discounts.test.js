import shoppingCartReducer from "../../src/shopping_cart_reducer"

it('should apply discount to the item', () => {
    const prevState = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 1,
                discount: 1
            }
        ],
        discount: 1
    }

    const discount = {
        discount: 0.5,
        id: "i2"
    }

    const real = shoppingCartReducer(prevState, {type: "APPLY_DISCOUNT_TO_ITEM", ...discount});

    const expected = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 1,
                discount: 0.5
            }
        ],
        discount: 1
    }

    expect(real).toEqual(expected);
});

it('applying discount to not existing item should change nothing', () => {
    const prevState = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 1,
                discount: 1
            }
        ],
        discount: 1
    }

    const discount = {
        discount: 0.5,
        id: "i3"
    }

    const real = shoppingCartReducer(prevState, {type: "APPLY_DISCOUNT_TO_ITEM", ...discount});

    const expected = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 1,
                discount: 1
            }
        ],
        discount: 1
    }

    expect(real).toEqual(expected);
});

it('should apply discount to the cart', () => {
    const prevState = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 1,
                discount: 1
            }
        ],
        discount: 1
    }

    const real = shoppingCartReducer(prevState, {type: "APPLY_DISCOUNT_TO_CART", discount: 0.75});

    const expected = {
        items: [
            {
                id: "i1",
                price: 100,
                count: 2,
                discount: 0.5
            },
            {
                id: "i2",
                price: 200,
                count: 1,
                discount: 1
            }
        ],
        discount: 0.75
    }

    expect(real).toEqual(expected);
});