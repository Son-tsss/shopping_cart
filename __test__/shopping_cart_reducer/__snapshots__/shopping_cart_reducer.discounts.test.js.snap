// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applying discount to not existing item should change nothing 1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 2,
      "discount": 0.5,
      "id": "i1",
      "price": 100,
    },
    Object {
      "count": 1,
      "discount": 1,
      "id": "i2",
      "price": 200,
    },
  ],
}
`;

exports[`should apply discount to the cart 1`] = `
Object {
  "discount": 0.75,
  "items": Array [
    Object {
      "count": 2,
      "discount": 0.5,
      "id": "i1",
      "price": 100,
    },
    Object {
      "count": 1,
      "discount": 1,
      "id": "i2",
      "price": 200,
    },
  ],
}
`;

exports[`should apply discount to the item 1`] = `
Object {
  "discount": 1,
  "items": Array [
    Object {
      "count": 2,
      "discount": 0.5,
      "id": "i1",
      "price": 100,
    },
    Object {
      "count": 1,
      "discount": 0.5,
      "id": "i2",
      "price": 200,
    },
  ],
}
`;
